runtime/interpreter/value.go

```diff
@@ -1940,6 +1940,7 @@ func (v *ArrayValue) iterate(
 			// atree.Array iteration provides low-level atree.Value,
 			// convert to high-level interpreter.Value
 			elementValue := MustConvertStoredValue(interpreter, element)
+			interpreter.checkInvalidatedResourceOrResourceReference(elementValue, locationRange)
```

Comment:
Is adding the calls to `checkInvalidatedResourceOrResourceReference` in the iteration functions (here for arrays, and below for dictionaries and composites) the main fix?

Reply:
Yes, essentially